src/stores/useReaderStore.ts

```diff
@@ -0,0 +1,79 @@
+import { create } from 'zustand';
+
+interface ScreenSettings {
+  fontSize: number;
+  zoomLevel: number;
+}
+
+interface ReaderState {
+  currentLine: number;
+  totalLines: number;
+  isLoading: boolean;
+  fontSize: number;
+  zoomLevel: number;
+  searchQuery: string;
+  guideContent: string[];
+  lastContentHash: string;
+  hasSetInitialPosition: boolean;
+  hasInitiallyScrolled: boolean;
+  userScrolling: boolean;
+}
+
+interface ReaderActions {
+  setCurrentLine: (line: number) => void;
+  setTotalLines: (total: number) => void;
+  setIsLoading: (loading: boolean) => void;
+  setFontSize: (size: number) => void;
+  setZoomLevel: (level: number) => void;
+  setSearchQuery: (query: string) => void;
+  setGuideContent: (lines: string[], contentHash: string) => void;
+  setHasSetInitialPosition: (value: boolean) => void;
+  setHasInitiallyScrolled: (value: boolean) => void;
+  setUserScrolling: (scrolling: boolean) => void;
+  resetReaderState: () => void;
+  updateScreenSettings: (screenId: string, settings: ScreenSettings) => void;
+}
+
+type ReaderStore = ReaderState & ReaderActions;
+
+const initialState: ReaderState = {
+  currentLine: 1,
+  totalLines: 0,
+  isLoading: true,
+  fontSize: 14,
```

Comment:
Anyway to have dedicated states for each overlay? 

eg fontSize and zoomLevel extracted to a named state